lib/shoulda/matchers/active_model/validate_length_of_matcher.rb

```diff
@@ -451,15 +456,19 @@ def allow_nil_does_not_match?
         end
 
         def allows_length_of?(length, message)
-          allows_value_of(string_of_length(length), message)
+          allows_value_of(element_of_length(length), message)
         end
 
         def disallows_length_of?(length, message)
-          disallows_value_of(string_of_length(length), message)
+          disallows_value_of(element_of_length(length), message)
+        end
+
+        def element_of_length(length)
```

Comment:
"Element" in this context sounds like it would be an element in an array, which isn't quite accurate. What do you think about `value_of_length`?